core/src/main/scala/io/qbeast/spark/index/OTreeDataAnalyzer.scala

```diff
@@ -216,6 +217,21 @@ object DoublePassOTreeDataAnalyzer
       indexStatus: IndexStatus,
       options: QbeastOptions): (DataFrame, TableChanges) = {
     logTrace(s"Begin: Analyzing the input data with existing revision: ${indexStatus.revision}")
+
+    // Check if the DataFrame is deterministic
+    logDebug(s"Checking the determinism of the input data")
+    val isSourceDeterministic =
+      analyzeDataFrameDeterminism(dataFrame, indexStatus.revision)
+    // TODO: we need to add columnStats control before the assert
```

Comment:
ColumnStats can't help the case if we consider the fact that it is not only the columns stats computed from the input dataframe will change, but the estimated cube domain sizes (consequently cube max weights) will also be wrong. 

In that case, the data can still be written but not as desired.

If we are ok with this, we can abort the process only when there is a `LinearTransformer` as other `Transformations` don't suffer from changing data.

Reply:
Now I see this comment, somehow related with the above. 

> Use revisionToUse instead of the existing revision in case we want to allow changing indexing columns

In addition to the reasoning, I think we can have a conflict there. 

Imagine the user inputs its own `columnStats`, and the source is non-deterministic (for `LinearTransformation`, as you mentioned). The DF analysis might output a min/max outside of `columnStats` boundaries. If that's the case, should we detect it and throw an error? Or should we proceed with indexing?